Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx

```diff
@@ -82,6 +82,33 @@ class SearchPanel extends React.Component<IhandleSearchChangeProps, ISearchPanel
     }
   }
 
+  componentDidUpdate(prevProps: IhandleSearchChangeProps) {
+    if (this.props.location !== prevProps.location) {
+      const querySrch = parse(this.props.location.search) as {
+        searchBy: string;
+        search: string;
+      };
+      const { searchBy, search } = querySrch;
+      // verify searchBy
+      if (searchBy) {
+        if (searchBy === "title") {
+          this.setState({ searchBy: SearchBy.Title });
+        } else {
+          this.setState({ searchBy: SearchBy.Genre });
+        }
+      } else {
+        this.setState({ searchBy: SearchBy.Title });
+      }
+      // verify search
+      if (search) {
+        this.setState({ value: search });
+      } else {
+        this.setState({ value: "" });
+      }
+    }
```

Comment:
overloaded with IFs. Make it easier, cleaner. whole code can be replaced with simple 2 lines 

```
this.setState({ 
   searchBy: searchBy === SearchBy.Title ? SearchBy.Title : SearchBy.Genre,
   value: search || ''
});
```

P.s. It's bad approach to call this.setState more then once in a row